test/bootstrapper.test.js

```diff
@@ -0,0 +1,80 @@
+var path = require('path');
+var loopback = require('loopback');
+
+var chai = require('chai');
+var dirtyChai = require('dirty-chai');
+var expect = chai.expect;
+chai.use(dirtyChai);
+
+var Bootstrapper = require('../lib/bootstrapper').Bootstrapper;
+
+describe('Bootstrapper', function() {
+  var app;
+  beforeEach(function() {
+    app = loopback();
+    process.bootFlags = [];
+  });
+
+  it('should honor options.phases', function(done) {
+    var options = {
+      app: app,
+      appRootDir: path.join(__dirname, './fixtures/simple-app'),
+      phases: ['load'],
+    };
+
+    var bootstrapper = new Bootstrapper(options);
+
+    var context = {
+      app: app,
+    };
+
+    bootstrapper.run(context, function(err) {
+      if (err) return done(err);
+      expect(context.configurations.application).to.be.an('object');
+      expect(context.configurations.bootScripts).to.be.an('array');
+      expect(context.configurations.middleware).to.be.an('object');
+      expect(context.configurations.models).to.be.an('object');
+      expect(context.configurations.tracker).to.eql('load');
+      expect(context.instructions).to.be.undefined();
```

Comment:
The assertions above will produce unhelpful errors on failure, because it will be difficult to tell whether which property did not pass the check.

A better solution:

```js
var cfgs = context.configurations;
expect(cfg.application, 'application').to.be.an('object');
// ...
expect(context.instructions, 'instructions').to.be.undefined();
```

The same comment applies to many tests below too.